### PR TITLE
Add validation for :path pseudo-header in HTTP/2 requests  RFC7540

### PR DESCRIPTION
**Pull request by arturobernalg**
This PR introduces validations for the :path pseudo-header field in accordance with the HTTP/2 specifications:

1. Ensures the :path pseudo-header field correctly represents the path and query parts of the target URI. Adheres to the "path-absolute" production. Optionally allows for a '?' character followed by the "query" production (as per Sections 3.3 and 3.4 of [[RFC3986](https://datatracker.ietf.org/doc/html/rfc3986)]).

2. Validates that requests in asterisk form have the value '*' for the :path pseudo-header field.

3. Enforces that the :path pseudo-header field is not left empty for "http" or "https" URIs.
4. Guarantees that "http" or "https" URIs without a path component include a '/' value for the :path pseudo-header field.
5.  Implements an exception for OPTIONS requests:
        If the request targets an "http" or "https" URI and lacks a path component, it must set the :path pseudo-header field to '*'.

### CODE REVIEW

File: httpcore5-h2/src/main/java/org/apache/hc/core5/http2/impl/DefaultH2RequestConverter.java [hunk 1]
```diff
@@ -208,4 +210,39 @@ public List<Header> convert(final HttpRequest message) throws HttpException {
...
+     * @param method The HTTP method of the request.
+     * @param scheme The scheme of the request.
+     * @param path The value of the {@code :path} pseudo-header field.
+     * @throws ProtocolException if the {@code :path} pseudo-header field is invalid.
+     */
+    private void validatePathPseudoHeader(final String method, final String scheme, final String path) throws ProtocolException {
+        if (URIScheme.HTTP.name().equalsIgnoreCase(scheme) || URIScheme.HTTPS.name().equalsIgnoreCase(scheme)) {
+            if (TextUtils.isBlank(path)) {
+                if (Method.OPTIONS.isSame(method)) {
+                    if (!"*".equals(path)) {
```
Comment (ok2c) [inline httpcore5-h2/src/main/java/org/apache/hc/core5/http2/impl/DefaultH2RequestConverter.java:234]: @arturobernalg This looks wrong. Should not the `path` already be known to be blank in this branch?

**Review by ok2c — CHANGES_REQUESTED** [review]
@arturobernalg What RFC have you been using as a reference for change?

**Comment by arturobernalg** [comment]
> @arturobernalg What RFC have you been using as a reference for change?

@ok2c 
this one [RFC7540](https://datatracker.ietf.org/doc/html/rfc7540#section-8.1.2.3)

File: httpcore5-h2/src/main/java/org/apache/hc/core5/http2/impl/DefaultH2RequestConverter.java [hunk 2]
```diff
@@ -208,4 +210,39 @@ public List<Header> convert(final HttpRequest message) throws HttpException {
...
+     * @throws ProtocolException if any of the validations fail.
+     */
+    private void validatePathPseudoHeader(final String method, final String scheme, final String path) throws ProtocolException {
+        if (URIScheme.HTTP.name().equalsIgnoreCase(scheme) || URIScheme.HTTPS.name().equalsIgnoreCase(scheme)) {
+            if (TextUtils.isBlank(path)) {
+                throw new ProtocolException("':path' pseudo-header field must not be empty for 'http' or 'https' URIs");
+            } else {
+                if (Method.OPTIONS.isSame(method)) {
+                    if (!"*".equals(path)) {
+                        throw new ProtocolException("OPTIONS request for an 'http' or 'https' URI without a path component must have a ':path' pseudo-header field with a value of '*'");
```
Comment (ok2c) [inline httpcore5-h2/src/main/java/org/apache/hc/core5/http2/impl/DefaultH2RequestConverter.java:237]: @arturobernalg This still looks wrong. Why should `OPTIONS` with the `/` path should be rejected as invalid? What you want to do it reject the path if it does not start with '/'  but make an exception for `OPTIONS` if the path is `*`.

**Review by ok2c — CHANGES_REQUESTED** [review]

File: httpcore5-h2/src/main/java/org/apache/hc/core5/http2/impl/DefaultH2RequestConverter.java (hunk 2 above)
Comment (arturobernalg) [reply httpcore5-h2/src/main/java/org/apache/hc/core5/http2/impl/DefaultH2RequestConverter.java]: @ok2c  please check again.

**Review by ok2c — APPROVED** [review]

**Pull request merged**